### PR TITLE
[6.4] UI port HostCollectionPackageManagement tests to airgun

### PR DESCRIPTION
**Pull request by ldjebran**
Depend on https://github.com/SatelliteQE/airgun/pull/189
with cdn False
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_hostcollection.py 
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 10 items                                                                                          2018-08-21 14:18:20 - conftest - DEBUG - BZ deselect is disabled in settings

collected 10 items                                                                                           

tests/foreman/ui_airgun/test_hostcollection.py::test_positive_create PASSED                            [ 10%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_edit PASSED                              [ 20%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_add_host PASSED                          [ 30%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_install_package PASSED                   [ 40%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_remove_package PASSED                    [ 50%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_upgrade_package PASSED                   [ 60%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_install_package_group PASSED             [ 70%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_remove_package_group PASSED              [ 80%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_install_errata PASSED                    [ 90%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_change_assigned_content PASSED           [100%]

======================================== 10 passed in 2595.52 seconds ========================================
```
with cdn True the only  (we need cdn True only for one test)

```bash
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest tests/foreman/ui_airgun/test_hostcollection.py -v -k "test_positive_change_assigned_content"
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 10 items                                                                                          2018-08-21 15:47:33 - conftest - DEBUG - BZ deselect is disabled in settings

collected 10 items / 9 deselected                                                                            

tests/foreman/ui_airgun/test_hostcollection.py::test_positive_change_assigned_content PASSED           [100%]

================================== 1 passed, 9 deselected in 653.93 seconds ==================================
```

### CODE REVIEW

File: tests/foreman/ui_airgun/test_hostcollection.py [hunk 1]
```diff
@@ -14,11 +14,162 @@
...
+        for host in vm_content_hosts
+    ]
+    host_collection = entities.HostCollection(
+        host=host_ids,
+        organization=module_org,
+    ).create()
+    return host_collection
+
+
+def _assert_package_installed(
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_hostcollection.py]: one of the things i like in pytest over unittest - you don't have to memorize all numerous assertions, python's builtin `assert` checks everything. Pls, let's not introduce extra assertions, but rather make this func `def package_installed` return bool so we can do `assert package_installed(...)` in tests.

File: tests/foreman/ui_airgun/test_hostcollection.py [hunk 2]
```diff
@@ -66,3 +243,262 @@ def test_positive_add_host(session):
...
+
+    :id: eead8392-0ffc-4062-b045-5d0252670775
+
+    :expectedresults: Package was successfully installed on all the hosts
+        in host collection
+
+    :CaseLevel: System
+    """
+    with session:
+        session.organization.select(org_name=module_org.name)
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_hostcollection.py]: redundant line here and in all the rest tests - module org get's selected by default, there's no point in extra selection.

File: tests/foreman/ui_airgun/test_hostcollection.py (hunk 1 above)
Comment (abalakh) [reply tests/foreman/ui_airgun/test_hostcollection.py]: Btw `assert not package_installed()` sounds way better to me than `_assert_package_installed(expected_installed=False)`

Comment (ldjebran) [reply tests/foreman/ui_airgun/test_hostcollection.py]: in this case we will wait for all retries and waist time

Comment (abalakh) [reply tests/foreman/ui_airgun/test_hostcollection.py]: For negative scenario? Imo in such case we should provide `retries`: `assert not package_installed(retries=0)`, that still sounds better to me.

**Comment by ldjebran** [comment]
@abalakh changes done
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_hostcollection.py 
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 10 items                                                                                          2018-09-04 11:50:44 - conftest - DEBUG - BZ deselect is disabled in settings

collected 10 items                                                                                           

tests/foreman/ui_airgun/test_hostcollection.py::test_positive_create PASSED                            [ 10%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_edit PASSED                              [ 20%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_add_host PASSED                          [ 30%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_install_package PASSED                   [ 40%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_remove_package PASSED                    [ 50%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_upgrade_package PASSED                   [ 60%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_install_package_group PASSED             [ 70%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_remove_package_group PASSED              [ 80%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_install_errata PASSED                    [ 90%]
tests/foreman/ui_airgun/test_hostcollection.py::test_positive_change_assigned_content PASSED           [100%]

======================================== 10 passed in 2804.57 seconds ========================================
```

**Comment by oshtaier** [comment]
pending two more cases

**Pull request merged**